src/YoutubeAtomPlayer.tsx

```diff
@@ -278,6 +288,25 @@ export const YoutubeAtomPlayer = ({
                     stateChangeListener,
                 );
 
+                /**
+                 * add a listener for when another youtube video is played in the page
+                 */
+                const youtubePlayerListener = (
+                    event: CustomEventInit<{ videoId: string }>,
+                ): void => {
+                    if (event instanceof CustomEvent) {
+                        if (event.detail.videoId !== videoId) {
+                            const playerStatePromise = player.current?.getPlayerState();
+                            playerStatePromise?.then((state) => {
+                                if (state === YT.PlayerState.PLAYING) {
+                                    player.current?.stopVideo();
+                                }
+                            });
+                        }
+                    }
+                };
+                document.addEventListener('video:play', youtubePlayerListener);
```

Comment:
Should this be in a useEffect so we can remove the eventListener when the component unmounts?